gravatar/src/main/java/com/gravatar/services/ProfileService.kt

```diff
@@ -99,58 +99,99 @@ public class ProfileService(okHttpClient: OkHttpClient? = null) {
 
     /**
      * Fetches a Gravatar profile for the given hash or username.
+     * This method throws any exception that occurs during the execution.
      *
      * @param hashOrUsername The hash or username to fetch the profile for
      * @return The fetched profile
      */
-    public suspend fun retrieveCatching(hashOrUsername: String): Result<Profile, ErrorType> = runCatchingService {
-        withContext(GravatarSdkDI.dispatcherIO) {
-            val response = service.getProfileById(hashOrUsername)
-            if (response.isSuccessful) {
-                val data = response.body()
-                if (data != null) {
-                    Result.Success(data)
-                } else {
-                    Result.Failure(ErrorType.UNKNOWN)
-                }
-            } else {
-                // Log the response body for debugging purposes if the response is not successful
-                Logger.w(
-                    LOG_TAG,
-                    "Network call unsuccessful trying to get Gravatar profile: $response.body",
-                )
-                Result.Failure(errorTypeFromHttpCode(response.code()))
-            }
+    public suspend fun retrieve(hashOrUsername: String): Profile = withContext(GravatarSdkDI.dispatcherIO) {
+        val response = service.getProfileById(hashOrUsername)
+        if (response.isSuccessful) {
+            response.body() ?: error("Response body is null")
```

Comment:
Quick question: 

Does `service.getProfileById(hashOrUsername)` always return a value even if the hash is not matched?  It will return an empty profile, right?

Reply:
The endpoint returns a `404` if the hash is not matched. We match that `404` to an empty state

<img width="1261" alt="image" src="https://github.com/user-attachments/assets/63ba83ed-4ab6-4b8f-af2e-4280a4ed483a">

Reply:
>We match that 404 to an empty state

Do we match that in the SDK? I can only see that in the demo app. 


I think we shouldn't throw an error if it is expected not to get a profile. It's probably better to return null instead.

Reply:
Updated. Let me know what you think.